azext_edge/edge/providers/orchestration/clone.py

```diff
@@ -0,0 +1,1419 @@
+        depends_on: Optional[Union[Iterable[str], str]] = None,
+        resource_group: Optional[str] = None,
+        subscription: Optional[str] = None,
+    ):
+        self.name = name
+        self.rcontainer_map: Dict[str, "ResourceContainer"] = {}
+        self.api_version = api_version
+        self.parameters = parameters
+        self.depends_on = depends_on
+        if isinstance(self.depends_on, str):
+            self.depends_on = {self.depends_on}
+        self.resource_group = resource_group
+        self.subscription = subscription
+
+    def add_resources(
+        self,
+        key: Union[StateResourceKey, str],
+        api_version: str,
+        data_iter: Iterable[dict],
+        depends_on: Optional[List[Union[StateResourceKey, str]]] = None,
+        config: Optional[Dict[str, Any]] = None,
+    ):
+        if isinstance(key, StateResourceKey):
+            key = key.value
+        depends_on = process_depends_on(depends_on)
+
+        to_enumerate = list(data_iter)
+
+        count = 0
+        for resource in to_enumerate:
+            count += 1
+            suffix = "" if count <= 1 else f"_{count}"
+            target_key = f"{key}{suffix}"
+
+            self.rcontainer_map[target_key] = ResourceContainer(
+                api_version=api_version,
+                resource_state=resource,
+                depends_on=depends_on,
+                config=config,
+            )
+
+    def get(self):
+        result = {
+            "type": "Microsoft.Resources/deployments",
+            "apiVersion": self.api_version,
+            "name": self.name,
+            "properties": {
+                "mode": "Incremental",
+                "template": {
+                    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+                    "contentVersion": "1.0.0.0",
+                    "resources": [r.get() for r in list(self.rcontainer_map.values())],
+                },
+            },
+        }
+        if self.resource_group:
+            result["resourceGroup"] = self.resource_group
+        if self.subscription:
+            # TODO: verify
+            # result["subscription"] = self.subscription
+            pass
+        if self.parameters:
+            input_param_map = {}
+            template_param_map = {}
+
+            for param in self.parameters:
+                target_value = (
+                    self.parameters[param]["value"]
+                    if "value" in self.parameters[param]
+                    else f"[parameters('{param}')]"
+                )
+                input_param_map[param] = {"value": target_value}
+                template_param_map[param] = {"type": self.parameters[param]["type"]}
+
+            result["properties"]["parameters"] = input_param_map
+            result["properties"]["template"]["parameters"] = template_param_map
+        if self.depends_on:
+            result["dependsOn"] = list(self.depends_on)
```

Comment:
note that depends_on can be an iterable of str or an str. And list(str) will just make a list of the characters in the str